modules/custom/cu_js_includes/pages/cu_js_includes.include_form.inc

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/**
+ * @file
+ * Form functions for cu_js_includes module.
+ */
+
+// Include confirm form file.
+module_load_include('inc', 'cu_js_includes', 'pages/cu_js_includes.confirm_form');
+
+/**
+ * Form for adding embeds.
+ *
+ * @param array $form
+ *   Empty form passed in to be built upon.
+ * @param array $form_state
+ *   State of form which might be in confirmation step.
+ * @param int|null $cu_js_includes_id
+ *   ID of embed if being edited or deleted.
+ *
+ * @return array
+ *   Fully built form ready to send to user.
+ */
+function cu_js_includes_form($form, &$form_state, $cu_js_includes_id = NULL) {
+
+  // If set to delete embeds, then return that form.
+  if (isset($form_state['storage']['confirm'])) {
+    return cu_js_includes_confirm($form, $form_state);
+  }
+
+  // Initialize default form variables.
+  $form = array();
+  $label = $path = '';
+  $data = array();
+  $cu_js_includes = NULL;
+  $id = NULL;
+  $size = variable_get('cu_js_includes_form_field_size', 128);
+
+  // Load embed if it exists.
+  if ($cu_js_includes_id != NULL && $cu_js_includes = entity_load_single('cu_js_includes', $cu_js_includes_id)) {
+    // Grab existing values to place in form.
+    $label = $cu_js_includes->name;
+    $type = $cu_js_includes->type;
+    $data = unserialize($cu_js_includes->data);
+    $id = $cu_js_includes_id;
+
+    // Store embed in form for form submission processing.
+    $form_state['cu_js_includes'] = $cu_js_includes;
+
+    // @todo Deal with embed templates entity not loading?
+  }
+  else {
+    // Get type.
+    $path_parts = explode('/', current_path());
+    $type = array_pop($path_parts);
+
+    // Get type definition.
+    $type_definition = cu_js_includes_get_include_type($type);
+  }
+
+  $form['include'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Include'),
+    '#weight' => -10,
+  );
+
+  $form['include']['label'] = array(
+    '#title' => t('Label'),
+    '#description' => t('Administrative label used in overview screen.'),
+    '#type' => 'textfield',
+    '#default_value' => $label,
+    '#required' => TRUE,
+    '#size' => $size,
+  );
+
+  // Add options for save and cancel.
+  $form['actions'] = array(
+    '#type' => 'actions',
+    'submit' => array(
+      '#type' => 'submit',
+      '#value' => t('Save'),
+    ),
+  );
+  // Add delete button only if editing an embed.
+  if (isset($cu_js_includes) && user_access('delete js includes')) {
+    $form['actions']['delete'] = array(
+      '#type' => 'submit',
+      '#submit' => array('cu_js_includes_form_submit'),
+      '#value' => t('Delete'),
+    );
+  }
+  $form['actions']['cancel'] = array(
+    '#type' => 'submit',
+    '#value' => t('Cancel'),
+    '#submit' => array('js_include_code_form_cancel'),
+    '#limit_validation_errors' => array(),
+  );
+
+
+
+  // @todo possibly disable parts of form if no "Publishing Options" are available
```

Comment:
I didn't see Publishing options in the schema. Can other code be simplified because we are not doing Publishing?